iothub/device/src/PayloadCollection.cs

```diff
@@ -133,6 +149,14 @@ public virtual string GetSerializedString()
             return Convention.PayloadSerializer.SerializeToString(Collection);
         }
 
+        /// <summary>
+        /// Remove all items from the collection.
+        /// </summary>
+        public void ClearCollection()
```

Comment:
Once a value is added to a `PayloadCollection`, it can only be updated but not removed. This is because property collection needs to be able to support `null` values for properties, so that service knows that those properties need to be removed: https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-device-twins#device-operations
For this reason we're adding a `Clear` method, that will allow the users to clear an entire collection and start afresh.

Reply:
If customers ask for "remove" functionality, we can add `RemoveRootProperty(string propertyName)`, `RemoveComponentProperty(string componentName, string propertyName)` and `RemoveComponent(string componentName)` methods.